tasks/codespell.rake

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+desc 'Run codespell over the files modified by the last commit'
+task :codespell do |_task|
+  next if Gem.win_platform?
+  next if ENV['CI'] # CI has its own workflow for this
+
+  sh 'which codespell &> /dev/null', verbose: false do |ok, _res|
+    if ok
+      sh 'git diff --name-only HEAD HEAD~1 | xargs codespell'
```

Comment:
This is probably too restrictive. It's possible to run `rake default` before committing which means there could be staged, or unstaged, changes that should be spellchecked. It's also possible to have multiple commits in a PR.

I think we should just call codespell on all files, like would happen in CI.

```suggestion
      sh 'git ls-files --empty-directory | xargs codespell'
```

This takes less than a second on my machine.

Reply:
I ran it with just `codespell` initially, which caused it to also run over `coverage`. Excluding that then also ran it over `.git`. I didn't think about doing it like CI does with misspell, that seems like a good idea.